src/gnb/ngap/nas.cpp

```diff
@@ -20,13 +20,52 @@
 #include <asn/ngap/ASN_NGAP_ProtocolIE-Field.h>
 #include <asn/ngap/ASN_NGAP_RerouteNASRequest.h>
 #include <asn/ngap/ASN_NGAP_UplinkNASTransport.h>
+#include <ue/nas/enc.hpp>
+#include "encode.hpp"
+#include <stdexcept>
 
 namespace nr::gnb
 {
 
-void NgapTask::handleInitialNasTransport(int ueId, const OctetString &nasPdu, int64_t rrcEstablishmentCause,
-                                         const std::optional<GutiMobileIdentity> &sTmsi)
+int32_t extractSliceInfoAndModifyPdu(OctetString &nasPdu) {
```

Comment:
Could you explain why the modification is needed for the original PDU

Reply:
Yes sure. In the current implementation, the UE is removing the information elements (including slice information), before sending the NAS PDU to gNB [ RemoveCleartextIEs() ]. gNB then selects one AMF blindly and forwards the request to the selected AMF. 

Now, to support intelligent AMF selection based on slice type, I have prevented removal of slice info from the PDU, when the UE sends it. Now, at the gNB end, the gNB uses this slice info., to select an appropriate AMF. Before forwarding the PDU to AMF, I had to remove the slice information from the PDU at gNB, else it leads to a "Unsupported Message Type received" error on AMF end.